### ISSUE TITLE
Design how to route messages to appropriate programs after transformation

### ISSUE THREAD

**Issue by mscaldas2012** [CDCgov/data-exchange-hl7#354]
We have a single HL7 Pipeline that can process any messages (so far only Case notifications being tested)

We need to design a way to route data for consumption. each message needs to be routed to the appropriate program. a program might receive data in multiple tables due to differences in schema from one condition to another or from historical data (old MMG version) to current MMGs.

## We need to route:
* Transformed messages - MMG-Based, MMG-SQL, Lake-of-Segments.
* Validation Reports - Structure and MMG validation reports (errors and warnings from both -ok and -err topics).
* Redaction Reports (once we have those).

## Assumptions:
* One program handles multiple conditions and multiple MMGs.
* A condition belongs to a single program.

# Routing proposal:

We should route message like this:
* For GenV2 messages with condition specific MMG - route to a table for that MMG
* For GenV2 generic messages (no condition specific MMG - MSH-21[3]), route to GenV2_{conditionCode} table. 
   * special scenario - if the condition Code belongs to a condition specific MMG, we could route that message to the table above, with only GenV2 info being populated.
* For GenV1 route to GenV1_Case_{conditionCode} or GenV1_Summary_{conditionCode}
* For legacy MMGs - route to a table for that MMG.

**Comment by mscaldas2012** [comment]
We had an initial meeting. Diagram used for discussion here: https://lucid.app/lucidchart/c756c505-7678-4964-ab00-8565062af81e/edit?viewport_loc=99%2C-188%2C2276%2C1065%2CddpaYJj16sIV&invitationId=inv_b7117784-e14d-439a-b716-c1aecb93df15

TL;DR> We are leaning towards dumping all transformed messages from Event Hub to a single Table in DBX.
DBX will then read that table and split messages belonging to different programs into different tables.

Open questions:
- [ ] Which Storage Account / Container should we use for the master table : DEX's or EDAV? (I though was EDAV, but we're using DEX)
- [ ] Which Storage Account / Container should we use for the program specific tables.?
- [ ] How to route basic GenV2 and GenV1 messages.
- [ ] What should be the program Table Granularity?
     - Full MMG Schema? (ex.: GenV2 + Hep Core + Hep A 1.0) -> multiple conditions on one table, multiple tables per program. single table with all data for one message.
     - Each MMG Schema( ex.: GenV2 is one Table, Hep Core another, Hep A 1.0 another, Hep A 1.1) -> message data in multiple tables, table with multiple program data (genV2)
     - Condition (ex.: 11088 in one table, 11080 in another table) -> multiple tables with similar schema, too many tables, programs will end up merging all those into one. Does not handle new MMG versions!
